yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -194,9 +195,17 @@ class TQueueExportTask
             .ValueOrThrow();
 
         auto transactionId = transaction->GetId();
-        WaitFor(transaction->LockNode(Queue_, ELockMode::Snapshot))
-            .ThrowOnError();
-        WaitFor(transaction->LockNode(ExportConfig_.ExportDirectory, ELockMode::Exclusive))
+        auto queueObjectId = WaitFor(transaction->LockNode(Queue_, ELockMode::Snapshot))
+            .ValueOrThrow()
+            .NodeId;
+
+        // Take it for guarantee that only one queue agent instance do this export.
+        WaitFor(transaction->LockNode(
```

Comment:
Let's add a test in which we take a shared lock with attribute_key on the export directory (to mimic what Logfeller does) and check that exports do not fail.